server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -135,6 +161,32 @@ class LivyServer extends Logging {
     sys.props("livy.server.serverUrl") = _serverUrl.get
   }
 
+  def runKinit(livyConf: LivyConf): Unit = {
+    val keytab = livyConf.get(LAUNCH_KERBEROS_KEYTAB)
+    val principal = livyConf.get(LAUNCH_KERBEROS_PRINCIPAL)
+    require(principal != null,
+      s"Kerberos requires ${LAUNCH_KERBEROS_KEYTAB.key} to be provided.")
+    require(keytab != null,
+      s"Kerberos requires ${LAUNCH_KERBEROS_PRINCIPAL.key} to be provided.")
+    val commands = Seq("kinit", "-kt", keytab, principal)
+    val proc = new ProcessBuilder(commands: _*).start()
```

Comment:
I thought about this yesterday and I'm slightly worried about this (running a child process without reading its output). kinit is not really noisy and its output shouldn't really fill the buffers, but you never know... it would be safer to just redirect the output (`ProcessBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT)`). 

You could do the same for the error stream but then you'd lose the ability to read the error and put it in the exception message. Which shouldn't be awful since the error will still be written to Livy's logs.

But at least stdout should be redirected.